cider/exec/module/batch/CiderBatchUtils.cpp

```diff
@@ -466,6 +461,36 @@ CiderBatch convertToArrowRepresentation(const CiderBatch& output_batch) {
       default:
         CIDER_THROW(CiderCompileException, "Type arrow convert not supported.");
     }
+    return 1;
+  }
+
+  ArrowArrayBuilder subBuilder;
+  int table_ptr_num = 0;
+  for (auto subType : type.struct_().types()) {
+    table_ptr_num += convertToArrowStruct(
+        subBuilder, subType, table_ptr, table_ptr_idx, table_row_num, column_num, names);
+    table_ptr_idx += table_ptr_num;
+  }
+  auto schema_and_array = subBuilder.build();
+  builder.addStructColumn(std::get<0>(schema_and_array), std::get<1>(schema_and_array));
+  return table_ptr_num;
+}
+
+CiderBatch convertToArrow(const CiderBatch& output_batch) {
+  std::shared_ptr<CiderTableSchema> table_schema = output_batch.schema();
+  auto column_num = table_schema->getFlattenColCount();
+  auto arrow_colum_num = output_batch.column_num();
+  CHECK_EQ(column_num * 2, arrow_colum_num);
+  auto table_row_num = output_batch.row_num();
+  ArrowArrayBuilder builder;
+  builder = builder.setRowNum(table_row_num);
+  const auto& types = table_schema->getColumnTypes();
+  const auto& names = table_schema->getFlattenColNames();
+  const int8_t** table_ptr = output_batch.table();
+  int table_ptr_idx = 0;
+  for (auto type : types) {
+    table_ptr_idx += convertToArrowStruct(
+        builder, type, table_ptr, table_ptr_idx, table_row_num, column_num, names);
   }
```

Comment:
Can we add some comments about the struct batch convert here? like how it's constructed with schema/table_ptr given.